session/session.go

```diff
@@ -514,6 +520,19 @@ func (s *session) isRetryableError(err error) bool {
 	return kv.IsRetryableError(err) || domain.ErrInfoSchemaChanged.Equal(err)
 }
 
+func (s *session) isTxnAborted(stmt sqlexec.Statement) error {
+	if s.txn.doNotCommit == nil {
+		return nil
+	}
+	if _, ok := stmt.(*executor.ExecStmt).StmtNode.(*ast.CommitStmt); ok {
```

Comment:
Please add comments to explain why the commit or rollback stmt can return nil.

Reply:
ok

Reply:
Could we add a type() method to the `sqlexec.Statement` interface later to avoid the type assertion, I think it is worth. But in this PR is ok.